packages/vm/src/runTx.ts

```diff
@@ -356,7 +346,7 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     if (opts.skipBalance === true && fromAccount.balance < maxCost) {
       // if skipBalance, ensure caller balance is enough to run transaction
       fromAccount.balance = maxCost
-      await this.stateManager.putAccount(caller, fromAccount)
+      await this.stateManager.putAccount(caller, fromAccount, true)
```

Comment:
Is there any false case here?

(so is the parameter necessary?)

Reply:
`putAccount` defaults to `touched = false`. This fixes edge cases like: generating canonical accounts where empty accounts are put in trie (note that if we have empty account at address A, where this account exists in trie B and not in trie C, we get a different state root). In previous versions where `putAccount` on EEI was called, this would touch the account (and `generateCanonicalGenesis` called this). This would mark these empty accounts as touched. If you now execute a tx and you do not touch these accounts, they get wiped in the end anyways.

Reply:
Hmm, from an API standpoint this is not so beautiful to have this extra `touched` option which puts the burden to think about what to put there also for more generic StateManager users, or keep too different method signatures, which is also not so nice.

I would think that it's worth to think about if we can just track these (very very few I guess) exceptions EVM internally in an inside data-structure `_nonTouched` or so and then also handle internally? 🤔

Reply:
Yep, this is the general idea also thus promoting the idea of moving this EVM logic  which now resides in the StateManager (before this PR, it was inside EEI) back into EVM to track it there and not let StateManager track this EVM logic.